ui/app/src/components/shared/StatusBadge.tsx

```diff
@@ -1,20 +1,81 @@
-import React from 'react';
+import { Callout, Stack, mergeStyleSets, FontWeights, Text, Icon } from '@fluentui/react';
+import React, { useState } from 'react';
 import { failedStates, inProgressStates, successStates } from '../../models/operation';
 
 interface StatusBadgeProps {
   status: string
+  resourceId?: string
 }
 
 export const StatusBadge: React.FunctionComponent<StatusBadgeProps> = (props: StatusBadgeProps) => {
+  const [showInfo, setShowInfo] = useState(false);
 
   let baseClass = "tre-badge";
-  if (inProgressStates.indexOf(props.status) !== -1) baseClass += " tre-badge-in-progress";
-  if (successStates.indexOf(props.status) !== -1) baseClass += " tre-badge-success";
-  if (failedStates.indexOf(props.status) !== -1) baseClass += " tre-badge-failed";
+  let badgeType = "";
+  if (props.status && successStates.indexOf(props.status) !== -1) { badgeType = "success"; }
+  else if (props.status && inProgressStates.indexOf(props.status) !== -1) { badgeType = "inProgress"; baseClass += " tre-badge-in-progress"; }
+  else if (props.status && failedStates.indexOf(props.status) !== -1) { badgeType = "failed"; baseClass += " tre-badge-failed"; }
 
-  return (
-    <>
-     {props.status && <span className={baseClass}>{props.status.replace("_", " ")}</span>}
-    </>
-  );
+  switch (badgeType) {
+    case "inProgress":
```

Comment:
We should consider using enums for these instead of strings like we have elsewhere - probs not relevant for this PR so don't worry about adding right now